File: 2019W25/preppindataw25.R
library(readxl)
library(readr)
library(dplyr)
library(tidyr)
library(splitstackshape)

final <- read_excel("E:/Wow _ PD data set.xlsx", 
                            sheet = "Gigs Data", col_types = c("numeric", 
                                                               "text", "date", "text", "text", "text")) %>%
  select(-'ConcertID') %>%
  {. ->> gigs} %>%
  .[!duplicated( gigs %>% mutate_all(., .funs=toupper)),] %>%
  mutate('All Artists' = if_else(grepl('\\/',`Concert`), `Concert`, `Artist`)) %>%
  cSplit(., 'All Artists', '/') %>% 
  gather(., key='var', value='Fellow Artist', starts_with('All Artists'),na.rm=TRUE) %>%
  select(-('var')) %>%
  mutate('Fellow Artist' = if_else(`Fellow Artist`==`Artist`, '', `Fellow Artist`)) %>%
  merge(.,home <- read_excel("E:/Wow _ PD data set.xlsx", 
                             sheet = "Home Locations") %>%
          rename('Home Longitude' = 'Longitude') %>%
          rename('Home Latitude' = 'Latitude'), by='Artist') %>%
  merge(.,loc <- read_csv("E:/LongLats.csv") %>%
          separate(`LongLats`, c('Longitude', 'Latitude'), sep = ',', convert= TRUE)
        , by='Location', all.x=TRUE)

View(final)

File: 2019W25/PreppindataW25.py
import pandas as pd
import re

gigs = pd.read_excel(pd.ExcelFile(r"E:\Wow _ PD data set.xlsx"),"Gigs Data") 
gigs['Concert Date'] = gigs['Concert Date'].apply(lambda x: x.date())
gigs.drop(['ConcertID'],axis=1,inplace=True)
gigs_ci = gigs.apply(lambda x: x.astype(str).str.lower()).drop_duplicates()  #1545 rows --> 1141 rows
gigs = gigs.loc[gigs_ci.index]
gigs = gigs.reset_index(drop=True)
gigs['All Artists']=gigs.apply(lambda x: x['Concert'] if re.search('\/',str(x['Concert'])) else x['Artist'],axis=1) 

home =  pd.read_excel(pd.ExcelFile(r"E:\Wow _ PD data set.xlsx"),"Home Locations")
home.rename(columns={'Longitude':'Home Longitude', 'Latitude':'Home Latitude'}, inplace = True)

loc = pd.read_csv("E:\LongLats.csv")
temp = loc['LongLats'].str.split(',', n = 1, expand = True)
loc['Longitude'] = temp[0].astype(float)
loc['Latitude'] = temp[1].astype(float)

final = pd.concat([gigs, 
                   pd.DataFrame([x for x in gigs['All Artists'].str.split('/').values.tolist()])], axis=1, sort=False)
final = final.melt(id_vars=[c for c in gigs.columns], value_name='Fellow Artists', var_name='To_drop')
final = final.dropna(subset=['Fellow Artists'])
final['Fellow Artists'] = final.apply(lambda x: '' if x['Fellow Artists'].strip()==x['Artist'] else x['Fellow Artists'].strip(), axis=1)
final = pd.merge(final, home, how='inner', on='Artist')
final = pd.merge(final, loc, how='left', on='Location')
final.drop(['All Artists','To_drop','LongLats'],axis=1,inplace=True)
